src/main/java/com/example/bookstore/dto/CreateBookRequestDto.java

```diff
@@ -1,12 +1,29 @@
 package com.example.bookstore.dto;
 
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Positive;
 import java.math.BigDecimal;
+import org.hibernate.validator.constraints.ISBN;
+import org.hibernate.validator.constraints.Length;
+import org.hibernate.validator.constraints.URL;
 
 public record CreateBookRequestDto(
+        @NotEmpty(message = "Title may not be empty")
```

Comment:
Is it for purpose you use NotEmpty but not NotBlank?

Reply:
I just want to test how it works, as I understand that NotEmpty is better to use with collections and NotBlank with strings, I replaced NotEmpty with NotBlank to use one annotation